build/build.sh

```diff
@@ -102,12 +102,35 @@ llvm-spirv)
 
     SPIRV_LLVM_TRANSLATOR_URL=https://github.com/KhronosGroup/SPIRV-LLVM-Translator.git
 
-    CMAKE_EXTRA_ARGS="-DLLVM_SPIRV_INCLUDE_TESTS=OFF -DSPIRV_SKIP_CLANG_BUILD=ON -DSPIRV_SKIP_DEBUG_INFO_TESTS=ON"
+    CMAKE_EXTRA_ARGS+=("-DLLVM_SPIRV_INCLUDE_TESTS=OFF" "-DSPIRV_SKIP_CLANG_BUILD=ON" "-DSPIRV_SKIP_DEBUG_INFO_TESTS=ON")
     LLVM_ENABLE_PROJECTS=""
     LLVM_ENABLE_RUNTIMES=
     NINJA_TARGET=install-llvm-spirv
     NINJA_TARGET_RUNTIMES=
     ;;
+rocm-*)
+    BASENAME=llvm-amdgpu
+    TAG=${VERSION}
+    ROCM_DEVICE_LIBS_BRANCH=${VERSION}
+
+    URL=https://github.com/RadeonOpenCompute/llvm-project.git
+    ROCM_DEVICE_LIBS_URL=https://github.com/RadeonOpenCompute/ROCm-Device-Libs.git
+
+    LLVM_ENABLE_PROJECTS="clang;lld;clang-tools-extra;compiler-rt"
+    CMAKE_EXTRA_ARGS+=("-DLLVM_TARGETS_TO_BUILD=AMDGPU;X86")
+    ;;
+amd-stg-open)
```

Comment:
As noted in another PR I think I'll need to change this to `rocm-trunk` or similar to keep in line with the other compilers. Else our things that detect "trunk" builds won't work (horrible to rely on this I know), but more importantly for me to know when two compilers are actually the same or not!

Reply:
Workign on it. I thnk the confusion here is we need a "trunk" build of the branch `amd-stg-open`, but a `rocm-{version}` branch for the canned builds, right?

Reply:
Yes. The release versions are tagged `rocm-{version}` and the public branch with the most recent changes is named `amd-stg-open`. I just used those names directly, but it certainly isn't obvious that they're related. It seems reasonable to use `rocm-trunk` and have that point to the branch `amd-stg-open`, if that's what you'd like to do.

Reply:
Awesome! thanks @cgmb - I've tried to make that so, but pre-coffee I had a few hiccups along the way (all of my own making). Will get this sorted though :) thanks for your patience!